serde/de.py

```diff
@@ -1129,18 +1129,18 @@ def {{func}}(cls=cls, maybe_generic=None, maybe_generic_type_vars=None, data=Non
   try:
     # create fake dict so we can reuse the normal render function
     {% if tagging.is_external() and is_taggable(t)  %}
-    ensure("{{typename(t)}}" in data , "'{{typename(t)}}' key is not present")
-    fake_dict = {"fake_key": data["{{typename(t)}}"]}
+    ensure("{{tagging.tag_for(t)}}" in data , "'{{tagging.tag_for(t)}}' key is not present")
+    fake_dict = {"fake_key": data["{{tagging.tag_for(t)}}"]}
```

Comment:
I wonder if we could inline the tags as literals, rather than looking them up during deserialization.